BitwardenShared/UI/Auth/StartRegistration/StartRegistrationView.swift

```diff
@@ -131,6 +132,37 @@ struct StartRegistrationView: View {
             .id(ViewIdentifier.StartRegistration.receiveMarketing)
         }
     }
+
+    /// The main content view that displays a scrollable layout of registration details.
+    ///
+    /// - Parameter proxy: A `GeometryProxy` instance that provides information about the size and
+    ///   coordinate space of the parent view.
+    /// - Returns: A `ScrollView` containing the logo image and registration details, with spacing
+    ///   and padding adjusted based on feature flag states.
+    ///
+    private func mainContent(with proxy: GeometryProxy) -> some View {
+        ScrollView(showsIndicators: false) {
+            VStack(spacing: 0) {
+                if store.state.isCreateAccountFeatureFlagEnabled {
+                    Spacer()
+
+                    Image(decorative: Asset.Images.logo)
+                        .resizable()
+                        .scaledToFit()
+                        .foregroundColor(Asset.Colors.iconSecondary.swiftUIColor)
+                        .frame(maxWidth: .infinity, maxHeight: 34)
+                        .padding(.horizontal, 12)
+
+                    Spacer()
+                }
+
+                registrationDetails
+            }
+            .padding([.horizontal, .vertical], 16)
+            .frame(minHeight: store.state.isCreateAccountFeatureFlagEnabled ? proxy.size.height : 0)
+        }
+        .frame(width: proxy.size.width)
+    }
```

Comment:
🎨 I might be tempted to include the `GeometryReader` in this function rather than pass it it. Alternatively, keep the `GeometryReader` in the body, and have this return the content that should be in the scroll view. It's minor, but I think it helps to keep the geometry reader closer to where it's used when possible. You could also use the `scrollView()` modifier to reduce one level of indentation and get the padding and background color as free defaults.

I also wonder if we should specify some minimum padding around the image for when the content grows.

```
        GeometryReader { proxy in
            mainContent()
                .padding(.vertical, 16)
                .frame(minHeight: store.state.isCreateAccountFeatureFlagEnabled ? proxy.size.height : 0)
                .scrollView(addVerticalPadding: false)
        }
```

```
    private func mainContent() -> some View {
        VStack(spacing: 0) {
            if store.state.isCreateAccountFeatureFlagEnabled {
                Spacer()

                Image(decorative: Asset.Images.logo)
                    .resizable()
                    .scaledToFit()
                    .foregroundColor(Asset.Colors.iconSecondary.swiftUIColor)
                    .frame(maxWidth: .infinity, maxHeight: 34)
                    .padding(.horizontal, 12)

                Spacer()
            }

            registrationDetails
        }
    }
```